### DISCUSSION TITLE
[Question] Enable test mode for Mailgun in settings.py

### DISCUSSION THREAD

**Discussion by rick020** [anymail/django-anymail#256] [Q&A] [answered]
According to the docs it is possible use the test mode in Mailgun by calling.
``` 
message = AnymailMessage(...)
message.esp_extra = {
    'o:testmode': 'yes',  # use Mailgun's test mode
}
```
Is it also possible to enable the test mode settings.py?

**Comment by medmunds** [comment, marked as answer]
Yes, you can use Anymail's [global send defaults](https://anymail.readthedocs.io/en/stable/sending/anymail_additions/#global-send-defaults) settings to provide default attributes for every message sent through Anymail:

```python
# settings.py
ANYMAIL = {
    ...
    "MAILGUN_SEND_DEFAULTS": {
        "esp_extra": {"o:testmode": "yes"},
    },
}
```
 
(Incidentally, Anymail's [Mailgun live integration tests](https://github.com/anymail/django-anymail/blob/dc0a46a815d062d52660b9237627b22f89093bce/tests/test_mailgun_integration.py#L23-L26) use this exact setting to help ensure they don't accidentally send real messages.)
